adr/2020_05_13_handling_non_crud_operations_in_api.md

```diff
@@ -0,0 +1,132 @@
+# Handling Non-CRUD operations in API
+
+* Status: proposed
+* Date: 2020-05-13
+
+## Context and Problem Statement
+
+Handling Non-CRUD operation over REST API is not trivial. Once operations are beyond Create, Read, Update, Delete, 
+which can be mapped directly to `POST`/`PUT`, `GET`, `PUT`/`PATCH`, `DELETE` HTTP methods there is no clear recommendation 
+how to map others actions. These actions include, but are not limited to, changes of states (described in the form of 
+workflow in a state machine) or command execution.
+
+## Decision Drivers
+
+* Solution should not be limited by its convention. We may need to support two different actions of the same name, 
```

Comment:
> We may need to support two different actions of the same name, but with different business logic.

I'm not sure whether I understand it correctly. Does it mean these two actions correspond to two different transitions leading to the same state?